src/sql-parser/src/parser.rs

```diff
@@ -5387,7 +5406,8 @@ impl<'a> Parser<'a> {
             Ok(JoinConstraint::On(constraint))
         } else if self.parse_keyword(USING) {
             let columns = self.parse_parenthesized_column_list(Mandatory)?;
-            Ok(JoinConstraint::Using(columns))
+            let alias = self.parse_join_using_alias(columns.clone())?;
```

Comment:
```suggestion
            let alias = self
                    .parse_keyword(AS)
                    .then(|| parser.parse_identifier())
                    .transpose()?;
```

And then you can remove the new fn.